gufe/tests/test_custom_json.py

```diff
@@ -184,6 +176,30 @@ def setup_method(self):
         ]
 
 
+class TestBytesCodec(CustomJSONCodingTest):
+    def setup_method(self):
+        self.codec = BYTES_CODEC
+        self.objs =[b'a test string']
+        self.dcts = [
+            {
+                ":is_custom:": True,
+                "__class__": "bytes",
+                "__module__": "builtins",
+                "latin-1": '(µ/ý \ri\x00\x00a test string'
+            }
+        ]
+
+    def test_legacy_uncompressed(self):
+        legacy_codec = JSONCodec(
+            cls=bytes,
+            to_dict=lambda obj: {"latin-1": obj.decode("latin-1")},
+            from_dict=lambda dct: dct["latin-1"].encode("latin-1")
+            )
```

Comment:
I can pull this out into a fixture if we want to test in more places, but I kept it simple for now.

Reply:
Good call; I went ahead and generalized this!